python/pyspark/taskcontext.py

```diff
@@ -162,7 +166,10 @@ def get(cls):
         running tasks.
 
         .. note:: Must be called on the worker, not the driver. Returns None if not initialized.
+            A RuntimeError will raise if it is not in a barrier stage.
         """
+        if not isinstance(cls._taskContext, BarrierTaskContext):
+            raise RuntimeError('''It is not in a barrier stage''')
```

Comment:
This is not supported behavior, so please raise `Exception` instead of `RuntimeError`

Reply:
Yeah, I think `Exception` is consistent at least.

Reply:
thanks for reviewing. updated